lib/navigation/control/consts.hpp

```diff
@@ -0,0 +1,76 @@
+#pragma once
+
+#include <array>
+#include <cstdint>
+#include <optional>
+
+#include <core/types.hpp>
+
+#if defined(__linux__)
+#include <eigen3/Eigen/Dense>
+//#else
+// TODO: add appropriate eigen include for mac here
+#endif
+
+namespace hyped::navigation {
+
+constexpr std::size_t state_dimension       = 3;
+constexpr std::size_t measurement_dimension = 1;
+constexpr std::size_t extended_dimension    = 2;
+// in order acc, velocity, displacement
+using StateVector           = Eigen::Matrix<core::Float, state_dimension, 1>;
+using StateTransitionMatrix = Eigen::Matrix<core::Float, state_dimension, state_dimension>;
+using StateTransitionCovarianceMatrix
+  = Eigen::Matrix<core::Float, state_dimension, state_dimension>;
+using ErrorCovarianceMatrix = Eigen::Matrix<core::Float, state_dimension, state_dimension>;
+using MeasurementMatrix
+  = Eigen::Matrix<core::Float, measurement_dimension, state_dimension>;  //[[1, 0, 0], [0, 0, 0],
+                                                                         //[0, 0, 0]]
+using MeasurementVector = Eigen::Matrix<core::Float, measurement_dimension, 1>;  //[acc_val, 0, 0]
+using MeasurementNoiseCovarianceMatrix
+  = Eigen::Matrix<core::Float, measurement_dimension, measurement_dimension>;
+using ExtendedStateVector
+  = Eigen::Matrix<core::Float, extended_dimension, 1>;  //[Jerk, Snap, Crackle, Pop, ...]
+using JacobianMatrix
+  = Eigen::Matrix<core::Float, extended_dimension, state_dimension>;  // higher order derivatives
+                                                                      // for propogation
+
+// TODOLater: move most of these to a config file
+static constexpr core::Float kTrackLength       = 100.0;  // m
+static constexpr core::Float kBrakingDistance   = 20.0;   // m TODOLater:check!
+static constexpr core::Float kPi                = 3.14159265359;
+static constexpr core::Float kWheelCicumference = kPi * 0.1;  // m TODOLater: check!
+static constexpr core::Float kStripeDistance    = 6.0;
+static constexpr bool kIsKeyenceActive          = true;
+
+// define sensor checks return type
+enum class SensorChecks { kUnacceptable = 0, kAcceptable };
+
+struct Quartiles {
```

Comment:
Aw I meant to fix it. It's because encoders preprocessor uses one and imu preprocessor uses the other. In reality they should probably get combined into one I think?